### PR TITLE
column level lineage

### PR DESCRIPTION
**Pull request by tobymao**

### CODE REVIEW

File: sqlglot/lineage.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+    Returns:
+        A lineage node.
+    """
+
+    expression = maybe_parse(sql, dialect=dialect)
+    optimized = optimize(expression, schema=schema)
+    scope = build_scope(optimized)
+
+    def to_node(column_name: str, scope: Scope, scope_name=None, upstream=None) -> Node:
```
Comment (GeorgeSittas) [inline sqlglot/lineage.py:53]: Types for `scope_name, `upstream`

File: sqlglot/lineage.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    return to_node(column if isinstance(column, str) else column.name, scope)
+
+
+class LineageHTML:
+    """Node to HTML generator using vis.js.
+
+    https://visjs.github.io/vis-network/docs/network/
+    """
+
+    def __init__(self, node: Node, dialect: t.Optional[str] = None, imports=True, **opts):
```
Comment (GeorgeSittas) [inline sqlglot/lineage.py:102]: Type for `import`

**Review by GeorgeSittas — APPROVED** [review]
Sweet!

File: sqlglot/lineage.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+        for node in node.walk():
+            if isinstance(node.expression, exp.Table):
+                label = f"FROM {node.expression.this}"
+                title = f"<pre>SELECT {node.name} FROM {node.expression.this}</pre>"
+                group = 1
+            else:
+                label = node.expression.sql(pretty=True, dialect=dialect)
+                source = node.source.transform(
+                    lambda n: exp.Tag(this=n, prefix="<b>", postfix="</b>")
+                    if n == node.expression
```
Comment (barakalon) [inline sqlglot/lineage.py:145]: I think this should be `n is node.expression`

**Review by barakalon — APPROVED** [review]

File: sqlglot/lineage.py [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+
+@dataclass(frozen=True)
+class Node:
+    name: str
+    expression: exp.Expression
+    source: exp.Expression
+    downstream: t.List[Node] = field(default_factory=list)
+
+    def walk(self) -> t.Generator[Node, None, None]:
```
[the remark this answers is not in the document]
Comment (tobymao) [reply sqlglot/lineage.py:18]: o rly

File: sqlglot/lineage.py (hunk 3 above)
Comment (tobymao) [reply sqlglot/lineage.py:145]: thanks, had that but reverted, going back

File: sqlglot/lineage.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (GeorgeSittas) [reply sqlglot/lineage.py:18]: I think there are 1 or 2 more instances of this type signature that can be simplified in the codebase.

**Pull request merged**